php-checks/src/main/java/org/sonar/php/checks/formatting/ControlStructureSpacingCheck.java

```diff
@@ -19,88 +19,157 @@
-    AstNode doubleArrow = foreachExpr.getFirstChild(PHPPunctuator.DOUBLEARROW);
+  private void checkForeachStatement(TokenVisitor tokenVisitor, ForEachStatementTree forloop, SyntaxToken asKeyword, SyntaxToken doubleArrow) {
 
-    boolean isSpaceCorrectAs = isSpaceAround(asKeyword, 1 /* space before*/, 1/* space after */);
-    boolean isSpaceCorrectDoubleArrow = doubleArrow == null || isSpaceAround(doubleArrow, 1/* space before*/, 1/* space after */);
+    boolean isSpaceCorrectAs = isExactlyOneSpaceAround(tokenVisitor, asKeyword);
+    boolean isSpaceCorrectDoubleArrow = doubleArrow == null || isExactlyOneSpaceAround(tokenVisitor, doubleArrow);
 
-    String keyword = null;
+    StringBuilder partialMessage = new StringBuilder();
     if (!isSpaceCorrectAs && isSpaceCorrectDoubleArrow) {
-      keyword = "\"" + asKeyword.getTokenOriginalValue() + "\"";
+      partialMessage.append("\"" + asKeyword.text() + "\"");
+
     } else if (isSpaceCorrectAs && !isSpaceCorrectDoubleArrow) {
-      keyword = "\"" + doubleArrow.getTokenOriginalValue() + "\"";
+      partialMessage.append("\"" + doubleArrow.text() + "\"");
+
     } else if (!isSpaceCorrectAs && !isSpaceCorrectDoubleArrow) {
-      keyword = "\"" + asKeyword.getTokenOriginalValue() + "\" and \"" + doubleArrow.getTokenOriginalValue() + "\"";
+      partialMessage.append("\"" + asKeyword.text() + "\" and \"" + doubleArrow.text() + "\"");
     }
 
-    if (keyword != null) {
-      formattingCheck.reportIssue("Put exactly one space after and before " + keyword + " in \"foreach\" statement.", node);
+    if (partialMessage.length() > 0) {
+      check.reportIssue(String.format(FOREACH_MESSAGE, partialMessage.toString()), forloop);
     }
   }
 
+  private static final boolean isExactlyOneSpaceAround(TokenVisitor tokenVisitor, SyntaxToken token) {
+    return TokenUtils.getNbSpaceBetween(tokenVisitor.prevToken(token), token) == 1
+      && TokenUtils.getNbSpaceBetween(token,tokenVisitor.nextToken(token)) == 1;
+  }
   /**
    * Check there is exactly one space after each ";" in for statement.
    */
-  private void checkSpaceForStatement(FormattingStandardCheck formattingCheck, AstNode node) {
+  private void checkSpaceForStatement(Tree tree) {
     boolean shouldReportIssue = false;
+    Iterator<Tree> it = ((PHPTree) tree).childrenIterator();
+    Tree next;
+    Tree previous = null;
 
-    for (AstNode semicolon : node.getChildren(PHPPunctuator.SEMICOLON)) {
-      Token semicolonToken = semicolon.getToken();
-      Token nextToken = semicolon.getNextAstNode().getToken();
-      int nbSpace = getNbSpaceBetween(semicolonToken, nextToken);
+    while (it.hasNext()) {
+      next = it.next();
 
-      if (nbSpace != 1 && isOnSameLine(semicolonToken, nextToken)) {
-        shouldReportIssue = true;
+      if (isSemicolon(previous)) {
+        SyntaxToken semicolonToken = (SyntaxToken) previous;
+        SyntaxToken nextToken = ((PHPTree) next).getFirstToken();
```

Comment:
@vilchik-elena no actually here i just forgot that iterator elements could be null. I think it should be checked.